src/scheduler.js

```diff
@@ -1,14 +1,21 @@
 /* eslint camelcase: 0 */
 
-// react platform is set to either react-dom or react-native during test and execution
-import { unstable_batchedUpdates } from 'react-platform'
 import { globalObj } from './utils'
+import { queue } from './queue'
 
 // this runs the passed function and delays all re-renders
 // until the function is finished running
+export let isInsideBatch = false
 export function batch (fn, ctx, args) {
   let result
-  unstable_batchedUpdates(() => (result = fn.apply(ctx, args)))
+  if (isInsideBatch) {
+    result = fn.apply(ctx, args)
+  } else {
+    isInsideBatch = true
+    result = fn.apply(ctx, args)
```

Comment:
Wrap this inside a try-finally block. `fn` runs user code (comp renders) which may throw errors. Not wrapping with try-finally will result in getting stuck in a "batch phase" until a not throwing batch runs.

Reply:
Maybe also cover this with a test.